### PR TITLE
remove duplicate query in receipts

### PR DESCRIPTION
**Pull request by fedekunze**
Closes: #126 

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: rpc/eth_api.go [hunk 1]
```diff
@@ -707,11 +709,15 @@ func (e *PublicEthAPI) GetTransactionReceipt(hash common.Hash) (map[string]inter
 	e.cliCtx.Codec.MustUnmarshalJSON(res, &logs)
 
 	txData := tx.TxResult.GetData()
+	fmt.Println(tx.TxResult.GetLog())
```
Comment (fedekunze) [inline rpc/eth_api.go]: `TxResult` logs seem to be fine:

```bash
[{"msg_index":0,"log":"executed EVM state transition; sender address 0x9e1Faa32b94dea3e38c48908ab9f6913d8d62151; contract address 0x3aD1Ad2396731C69e9cEF0af102EfBc3a1a4B35A","events":[{"type":"ethereum","attributes":[{"key":"amount","value":"0"}]},{"type":"message","attributes":[{"key":"action","value":"ethereum"},{"key":"module","value":"evm"},{"key":"sender","value":"0x9e1Faa32b94dea3e38c48908ab9f6913d8d62151"}]}]}]
```

File: rpc/eth_api.go [hunk 2]
```diff
@@ -301,13 +301,15 @@ func (e *PublicEthAPI) SendTransaction(args params.SendTxArgs) (common.Hash, err
...
 
-	// Broadcast transaction
+	// Broadcast transaction in sync mode (default)
 	res, err := e.cliCtx.BroadcastTx(txBytes)
 	// If error is encountered on the node, the broadcast will not return an error
 	if err != nil {
 		return common.Hash{}, err
 	}
 
+	fmt.Println(res)
```
Comment (fedekunze) [inline rpc/eth_api.go]:
```
Response:
  TxHash: F4F00B3734E28D6DA078CCBE09DAE8465D28EDD650E26E6A44953C3DECB51CCE
  Raw Log: []
  Logs: []
```

File: rpc/eth_api.go [hunk 3]
```diff
@@ -698,37 +700,39 @@ func (e *PublicEthAPI) GetTransactionReceipt(hash common.Hash) (map[string]inter
 		status = hexutil.Uint(0)
 	}
 
-	res, _, err := e.cliCtx.Query(fmt.Sprintf("custom/%s/%s/%s", types.ModuleName, evm.QueryTxLogs, hash.Hex()))
```
Comment (fedekunze) [inline rpc/eth_api.go:701]: this query was redundant

File: rpc/eth_api.go [hunk 4]
```diff
@@ -698,37 +700,39 @@ func (e *PublicEthAPI) GetTransactionReceipt(hash common.Hash) (map[string]inter
...
-
 	txData := tx.TxResult.GetData()
+	fmt.Println(tx.TxResult.GetLog())
+
 	data, err := types.DecodeResultData(txData)
 	if err != nil {
 		return nil, err
 	}
 
+	fmt.Println(data.String())
```
Comment (noot) [inline rpc/eth_api.go]: I'm seeing `internal error` coming from this when deploying a contract through rpc_test.go `TestEth_GetTransactionReceipt`

Comment (fedekunze) [reply rpc/eth_api.go]: that's probably out of gas

File: rpc/eth_api.go [hunk 5]
```diff
@@ -698,38 +698,36 @@ func (e *PublicEthAPI) GetTransactionReceipt(hash common.Hash) (map[string]inter
...
-	e.cliCtx.Codec.MustUnmarshalJSON(res, &logs)
-
 	txData := tx.TxResult.GetData()
 
 	data, err := types.DecodeResultData(txData)
 	if err != nil {
 		status = 0 // transaction failed
 	}
 
+	// TODO: consider changing to geth's Receipt struct
```
Comment (noot) [inline rpc/eth_api.go]: geth's eth_getTransactionReceipt actually returns a `map[string]interface{}` as well 
https://github.com/ethereum/go-ethereum/blob/b9df7ecdc3d3685180ceb29665bab59e9f614da5/internal/ethapi/api.go#L1337

their Receipt struct doesn't have all the info we want, so I think leaving it like this is fine

**Review by noot — APPROVED** [review]
looks good, I think the TODO can probably be removed 

also please lint!

**Pull request merged**